### ISSUE TITLE
Assigning to a `resize_`ed slice is broken

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#4979]
The following code
```
import torch
a = torch.ones(5, 3)
y = a[:,1].resize_(1, 5)
y[0] = torch.arange(0, 5)
```
produces the following output for `a`:
```
1  0  1
2  3  4
1  1  1
1  1  1
1  1  1
```
But, as the user, you'd really expect `a` to look like:
```
1  0  1
1  1  1
1  2  1
1  3  1
1  4  1
```
I'll take care of this.

**Comment by fmassa** [comment]
I'm not sure we want to modify the current behaviour of reshape.

The current contract is that if the resize operation has happened, then the resized tensor is contiguous (except if the tensor already has the right shape, in which case the resize is a no-op).

I'm afraid that modifying this code might break code in a bunch of places.

**Comment by zou3519** [comment]
Ah I see. That's a good point and I agree with you that changing the behavior would be unwise.

**Comment by albanD** [comment]
@zou3519 are you sure that you didn't mean to use `view` instead of `resize_` ?
Using `view`, gives the output you expect.

**Comment by zou3519** [comment]
@albanD thanks! I was trying to use `resize_` because I was emulating how the internals of `torch.sum(a, b, dim, out=c)` works (`c` gets `resize_`ed into the correct shape, which can lead to some weird things). But thank you for pointing out that `view` is the correct way to do it in this case
